src/main/java/model/Rank.java

```diff
@@ -0,0 +1,41 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public enum Rank {
+    FIRST(6, 2000000000),
+    SECOND(5, 30000000),
+    THIRD(5, 1500000),
+    FOURTH(4, 50000),
+    FIFTH(3, 5000),
+    NONE(0, 0);
+
+    private final int matchCount;
+    private final int winningMoney;
+
+    Rank(int matchCount, int winningMoney) {
+        this.matchCount = matchCount;
+        this.winningMoney = winningMoney;
+    }
+
+    public int getMatchCount() {
+        return matchCount;
+    }
+
+    public int getWinningMoney() {
+        return winningMoney;
+    }
+
+    public static Rank getRank(int matchCount, boolean hasBonusNo) {
+        ArrayList<Rank> ranks = new ArrayList<>(Arrays.asList(Rank.values()));
+        if (!hasBonusNo) {
+            ranks.remove(Rank.SECOND);
+        }
+        return ranks.stream()
+                .filter(rank -> rank.matchCount == matchCount)
+                .findFirst()
+                .orElseGet(() -> Rank.NONE);
```

Comment:
```suggestion
               .orElse(Rank.NONE);
```

인텔리제이도 추천하고 있지만, 위와 같이 변경해보면 어떨까 싶어요!   
단지 ``orElse``를 사용할 때는 주의가 필요하다는 점 [참고자료](http://homoefficio.github.io/2019/10/03/Java-Optional-%EB%B0%94%EB%A5%B4%EA%B2%8C-%EC%93%B0%EA%B8%B0/) 의 2번 주제를 읽어보시면 좋을 것 같아요! 
겸사겸사 참고자료의 다른 문항들도 같이 읽어보고 Optional을 바르게 쓰는 방법을 숙지하면 좋을 것 같아요 😉

왜 제가 이렇게 고치라고 했는지에 대해서도 코멘트에 추가로 코멘트 달아주시면 좋을 것 같아요 ~!

Reply:
이미 생성되었거나 계산된 값은 `orElse`을 사용하는 것이 좋다고 되어있었습니다. 살펴보면서 고민을 해봤더니, `Rank.NONE` 자체는 새롭게 메모리에 할당되어 만들어지는 것이 아닌 이미 존재하는 상수이기 때문에 이미 존재하는 상수를 가져오기 위해 `Supplier`를 가져오는 것은 비효율적이라는 결론에 도달했습니다.